src/core/mac/sub_mac_wed.cpp

```diff
@@ -84,6 +101,30 @@ void SubMac::HandleWedTimer(void)
     }
 }
 
+void SubMac::HandleWedReceiveOrSleep(void)
+{
+    uint32_t interval;
+    int32_t  delay;
+
+    mIsWedSampling = !mIsWedSampling;
+    interval       = mIsWedSampling ? mWakeupListenDuration : (mWakeupListenInterval - mWakeupListenDuration);
+    delay          = mIsWedSampling ? kMinReceiveOnAfter : -kMinReceiveOnAhead;
+
+    mWedSampleTime += interval;
+    mWedTimer.FireAt(mWedSampleTime + delay);
```

Comment:
I think this would be more readable.
```suggestion
    TimeMilli fireTime;

    mIsWedSampling = !mIsWedSampling;

    if (mIsWedSampling)
    {
        fireTime = mWedSampleTime + mWakeupListenDuration + kMinReceiveOnAfter;
    } else {
        mWedSampleTime += mWakeupListenInterval;
        fireTime = mWedSampleTime - kMinReceiveOnAhead;
    }

    mWedTimer.FireAt(fireTime);
```